src/Compilers/CSharp/Test/Syntax/Parsing/StatementParsingTests.cs

```diff
@@ -4252,6 +4252,491 @@ public void ParseNewNullableWithInitializer()
+                                        N(SyntaxKind.InvocationExpression);
+                                        {
+                                            N(SyntaxKind.SimpleMemberAccessExpression);
+                                            {
+                                                N(SyntaxKind.IdentifierName);
+                                                {
+                                                    N(SyntaxKind.IdentifierToken, "Console");
+                                                }
+                                                N(SyntaxKind.DotToken);
+                                                N(SyntaxKind.IdentifierName);
+                                                {
+                                                    N(SyntaxKind.IdentifierToken, "WriteLine");
+                                                }
+                                            }
+                                            N(SyntaxKind.ArgumentList);
+                                            {
+                                                N(SyntaxKind.OpenParenToken);
+                                                N(SyntaxKind.Argument);
+                                                {
+                                                    N(SyntaxKind.StringLiteralExpression);
+                                                    {
+                                                        N(SyntaxKind.StringLiteralToken, "\"answer\"");
+                                                    }
+                                                }
+                                                N(SyntaxKind.CloseParenToken);
+                                            }
+                                        }
+                                        N(SyntaxKind.SemicolonToken);
+                                    }
+                                    N(SyntaxKind.BreakStatement);
+                                    {
+                                        N(SyntaxKind.BreakKeyword);
+                                        N(SyntaxKind.SemicolonToken);
+                                    }
+                                }
+                                N(SyntaxKind.CloseBraceToken);
+                            }
+                            N(SyntaxKind.CloseBraceToken);
+                        }
+                    }
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact, WorkItem(66971, "https://github.com/dotnet/roslyn/issues/66971")]
+        public void ParseErrantStatementInCase2()
+        {
+            UsingTree("""
+                class C
+                {
+                    void M()
+                    {
+                        switch (new object())
+                        {
+                            bool SomeTest() => o is 42;
```

Comment:
I definitely think the changed handling of 'case' outside switch puts us in a better place.

However, I would still find this error a little esoteric as a new user. The compiler is acting like I meant to end the switch, but then it's telling me to start up a new switch? What? It would be easier for me if it just gave an error on the first token of the statement that said 'case' was expected, for example, or an error that said statements in general can't go directly inside a switch, they need to be inside a 'case block' or some such term.

Reply:
I hear you, and i'm willing to invest in that direciton a little more if you have strong feelings on that.  *however* i will say that this is how things work with other list parsing.  If lists run into an elmeent they can't handle, but a parent can, they just terminate themselves (stating what is missing) and let the parent then handle the next.

So this fits/follows the model we have everywhere elsewhere.  So i'm not really wanting to deviate unless we followed this policy for all the other cases.  

Thoughts/

Reply:
I think this handling is a good improvement, and we should take this before starting to invest more in the space.

Reply:
Fwiw I don't see this as a change that requires rethinking how errors are handled everywhere. I think while there are standard patterns for recovering from syntax errors there are still numerous ad-hoc decisions made on how to interpret misplaced code and that's OK. For example, we handle 'int arr[0];' as a single statement with bad tokens embedded after 'arr' instead of, say, adding a missing semicolon after 'arr', then parsing [0] as a bad attribute list or something. People will tend to put statements into things that look like blocks which is why we're interested in recovering from errant ; in switch expressions, object creation and so on.

All that said, I'm completely satisfied to take the PR as is. If you decide to do any additional work to extend the error recovery here then great. Otherwise we can just look at it again if further feedback comes in from the community.